build_tfm.py

```diff
@@ -109,11 +109,6 @@ def _get_target_info(target, toolchain=None):
     else:
         tc = TARGET_MAP[target].tfm_default_toolchain
 
-    global TC_DICT
-    if not TOOLCHAIN_CLASSES[TC_DICT.get(tc)].check_executable():
-        msg = "Toolchain %s was not found in PATH" % tc
-        raise Exception(msg)
-
```

Comment:
We are removing the toolchain check, because this PR changes "ARMC6" to "ARM" as required by `mbed-tools`, but `check_executable()` still assumes "ARM" to be Arm Compiler 5.

@ARMmbed/mbed-b-tools  Is there a plan to update `check_executable()`? Or is it for the legacy tools only?

Reply:
https://github.com/ARMmbed/mbed-os/blob/26606218ad9d1ee1c8781aa73774fd7ea3a7658e/tools/toolchains/arm.py#L67-L71